### ISSUE TITLE
Make chain call function accessible for users

### ISSUE THREAD

**Issue by iszulcdeepsense** [TheRacetrack/racetrack#156]
In the  [sample with job chaining](https://github.com/TheRacetrack/plugin-python-job-type/blob/master/sample/python-chain/entrypoint.py#L19) it is shown how to use `call_fatman` to chain to another job.

Since 
1. multiple developers has to do chaining
2. the way in which chaining is done has been changing over time (a possibly still will be) and it is the racetrack developers who knows how `call_fatman` should be changing accordingly
3. model code becomes deprecated as racetrack is being developed

Why not make this function available to users of racetrack to import or build it in automatically?
Can we avoid duplication and copy-pasting the same snippet to the source code of a job?

**Comment by JosefAssadERST** [comment]
What's te main challenge we're trying to solve for the job developers here? Assuming they know the fatman name and path(discoverable from RT) ten it's mainly the internal name of PUB isn't it?

If that's correct I agree it's a challenge, but maybe not a big one?

**Comment by iszulcdeepsense** [comment]
I think the challenges are to:
- implement authentication: get auth token from env and put it to the auth header - it's been changed once the new auth model came in,
- prepare request tracing ID for tracking the requests chains, eg. in OpenTelemetry or logs. This is not required, but if a developer forgets about it, loses the chance to track the chain of calls.
- find out PUB's internal hostname
- build the PUB's path - that's been changed already due to changing the URL format. And it's gonna be changed again soon once we replace `/pub/fatman/...` to `/pub/job/...`.

However, the most severe challenge is to keep up with the evolving Racetrack. I'd like to prevent job developers from frequent updating their jobs due to internal changes in Racetrack. 

Maybe they could use some kind of interface? But what if some day we have to change the interface?
I don't have any decent solution for now, this issue aims rather at exploring that. This issue was originally raised by one of developers in a former issues tracker.

**Comment by JosefAssadERST** [comment]
Yeah. There's a fair bit of design work and thinking in this one.

**Comment by iszulcdeepsense** [comment]
@JosefAssadERST any opinions about this?

**Comment by iszulcdeepsense** [comment]
Now I'm more inclined to put the "chain call snippet" to a separate, tiny PyPI library, eg. "racetrack-chain-call". It makes jobs more dependent on Racetrack, but only those that were already coupled with Racetrack by making chain calls inside, thus requiring Racetrack to be present.

**Comment by JosefAssadERST** [comment]
How about shipping it as something importable in the racetrack-client tool, so it's also usable as a library?

**Comment by JosefAssadERST** [comment]
> racetrack-client was supposed to be language-agnostic core of the Racetrack, but it's getting related to Python3 plugin

Very good point. That strongly discourages me from my own idea.
